stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -528,7 +536,13 @@ def _apply_on_move(self, move, assign_picking=True):
             return False
         logger.info("Applying flow '%s' on '%s'", self.name, move)
         rule = self._get_rule_from_delivery_route()
```

Comment:
As now the flow can be re-run multiple times (as we keep a copy of the default picking type), we should not re-run if we don't apply a new flow. Otherwise each time you reassign the move to a new picking which is awkward and useless
```suggestion
        rule = self._get_rule_from_delivery_route()
        if move.rule_id == rule:
            return
        logger.info("Applying flow '%s' on '%s'", self.name, move)
```